src/apps/competitions/tasks.py

```diff
@@ -98,58 +120,44 @@ def run_submission(submission_pk, is_scoring=False):
-                new_details.data_file.save(f'{detail_name}.txt', ContentFile(''.encode()))  # must encode here for GCS
-                run_arguments[detail_name] = make_url_sassy(new_details.data_file.name, permission="w")
-
-            print("Task data:")
-            print(run_arguments)
-
-            # Pad timelimit so worker has time to cleanup
-            time_padding = 60 * 20  # 20 minutes
-            time_limit = submission.phase.execution_time_limit + time_padding
-
-            task = app.send_task('compute_worker_run', args=(run_arguments,), queue='compute-worker',
-                                 soft_time_limit=time_limit)
-            submission.task_id = task.id
-            submission.status = Submission.SUBMITTED
-            submission.save()
+                    if task.input_data:
+                        run_arguments['input_data'] = make_url_sassy(task.input_data.data_file.name)
 
+            if is_scoring and task.reference_data:
+                run_arguments['reference_data'] = make_url_sassy(task.reference_data.data_file.name)
+
+            run_arguments['program_data'] = make_url_sassy(
+                path=submission.data.data_file.name if not is_scoring else task.scoring_program.data_file.name
+            )
+            run_arguments['result'] = make_url_sassy(
+                path=submission.result.name,
+                permission='w' if not is_scoring else 'r'
+            )
+            detail_names = SubmissionDetails.DETAILED_OUTPUT_NAMES_PREDICTION if not is_scoring else SubmissionDetails.DETAILED_OUTPUT_NAMES_SCORING
+            for detail_name in detail_names:
+                run_arguments[detail_name] = create_detailed_output_file(detail_name, submission)
+            send(submission, run_arguments)
     else:
-        if not is_scoring:
-            # Pre-generate file path by setting empty file here
-            submission.result.save('result.zip', ContentFile(''.encode()))  # must encode here for GCS
-            # Run the submission
-            run_arguments["program_data"] = make_url_sassy(submission.data.data_file.name)
-            run_arguments["result"] = make_url_sassy(submission.result.name, permission='w')
-        else:
-            # Run the scoring_program
-            run_arguments["program_data"] = make_url_sassy(submission.phase.scoring_program.data_file.name)
-            run_arguments["result"] = make_url_sassy(submission.result.name)
-            # run_arguments["ingestion_program"] = make_url_sassy(submission.phase.ingestion_program.data_file.name)
+        run_arguments["program_data"] = make_url_sassy(
+            path=submission.data.data_file.name if not is_scoring else submission.phase.scoring_program.data_file.name
+        )
+        run_arguments["result"] = make_url_sassy(
+            path=submission.result.name,
+            permission='w' if not is_scoring else 'r'
+        )
 
```

Comment:
Still not super-duper happy with how this DRY refactor turned out. But calling it good enough until we do a full refactor for packaging all phases as tasks.

Reply:
👍